hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java

```diff
@@ -54,8 +77,169 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         }
     }
 
+    @Override
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        requireNonNull(txn);
+        final var op = txn.tokenMintOrThrow();
+        validateTruePreCheck(op.hasToken(), INVALID_TOKEN_ID);
+        validateFalsePreCheck(!op.metadata().isEmpty() && op.amount() > 0, INVALID_TRANSACTION_BODY);
+        validateFalsePreCheck(op.amount() < 0, INVALID_TOKEN_MINT_AMOUNT);
+    }
+
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
-        throw new UnsupportedOperationException("Not implemented");
+        final var op = context.body().tokenMintOrThrow();
+        final var tokenId = context.body().tokenMintOrThrow().tokenOrThrow();
+
+        validateSemantics(context);
+
+        final var tokenStore = context.writableStore(WritableTokenStore.class);
+        final var tokenRelStore = context.writableStore(WritableTokenRelationStore.class);
+        final var accountStore = context.writableStore(WritableAccountStore.class);
+        // validate token exists
+        final var token = tokenStore.get(tokenId);
+        validateTrue(token != null, INVALID_TOKEN_ID);
+        // get the config needed for validation
+        final var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
+        final var maxAllowedMints = tokensConfig.nftsMaxAllowedMints();
```

Comment:
Looks like this could be moved into the NFT block on line ~114, but it's fine to keep here too

Reply:
Done.